doc/guides/optimizations/footprint.rst

```diff
@@ -0,0 +1,55 @@
+.. _footprint:
+
+Optimizing for Footprint
+########################
+
+Stack Sizes
+***********
+
+Stack sizes of various system threads are sepcified generously to allow for
+usage in different scenarios on as many supported platforms as possible. You
+should start the optimization process by reviewing all stack sizes and adjusting
+them for your application:
+
+CONFIG_ISR_STACK_SIZE
+  Set to 2048 by default
+
+CONFIG_MAIN_STACK_SIZE
+  Set to 1024 by default
+
+CONFIG_IDLE_STACK_SIZE
+  Set to 320 by default
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE
+  Set to 1024 by default
+
```

Comment:
OFFLOAD_WORKQUEUE_STACK_SIZE, too

And, perhaps, PRIVILEGED_STACK_SIZE, if you care about user mode